src/proto/mod.rs

```diff
@@ -1,6 +1,5 @@
-use crate::FaktoryError;
+use crate::error::{Connect, Error, Protocol};
```

Comment:
nit: my personal preference here would be to just `use crate::error::{self, Error}`, and then below write `error::Connect` to make it a little clearer at a glance. Not a blocking concern though!